src/Stack/Types/Config.hs

```diff
@@ -377,6 +381,7 @@ instance FromJSON Resolver where
 renderResolver :: Resolver -> Text
 renderResolver (ResolverSnapshot name) = renderSnapName name
 renderResolver (ResolverGhc (MajorVersion x y)) = T.pack $ concat ["ghc-", show x, ".", show y]
+renderResolver (ResolverCustom url) = "custom: " <> url
```

Comment:
It's going to be `resolver: "custom: ..."`. I originally did the same `custom-` kind of prefix as everything else, but it looks too weird to say "custom-https://..."